File: core/model/posts/domain/post/Post.cs
using System;
using System.Collections.Generic;
using SocialNetwork.core.model.players.domain;
using SocialNetwork.core.model.posts.domain.comment;
using SocialNetwork.core.model.posts.domain.reaction;
using SocialNetwork.core.model.shared;

namespace SocialNetwork.core.model.posts.domain.post
{
    public class Post : Entity<PostId>, IAggregateRoot
    {
        public TextBox PostText { get; private set; }

        public PlayerId PlayerCreator { get; private set; }

        public List<Tag> Tags { get; private set; }

        public List<Reaction> Reactions { get; private set; }

        public List<Comment> Comments { get; private set; }

        public CreationDate CreationDate { get; private set; }

        protected Post()
        {
            // for ORM
        }

        protected Post(PostId id, TextBox postText, Player playerCreator, List<Tag> tagsList, List<Reaction> reactionsList,
            List<Comment> commentsList, CreationDate creationDate)
        {
            this.Id = id;
            this.PostText = postText;
            this.PlayerCreator = playerCreator.Id;
            this.Tags = new(tagsList);
            this.Reactions = new(reactionsList);
            this.Comments = new(commentsList);
            this.CreationDate = creationDate;
        }

        public Post(TextBox postText, Player playerCreator, List<Tag> tagsList)
        {
            this.Id = new PostId(Guid.NewGuid());
            this.PostText = postText;
            this.PlayerCreator = playerCreator.Id;
            this.Tags = new(tagsList);
            this.Reactions = new();
            this.Comments = new();
            this.CreationDate = new();
        }

        public Post(TextBox postText, Player playerCreator)
        {
            this.Id = new PostId(Guid.NewGuid());
            this.PostText = postText;
            this.PlayerCreator = playerCreator.Id;
            this.Tags = new();
            this.Reactions = new();
            this.Comments = new();
            this.CreationDate = new();
        }

        public bool AssignTag(Tag newTag)
        {
            if (this.Tags.Contains(newTag))
                return false;

            this.Tags.Add(newTag);
            return true;
        }

        public bool RemoveTag(Tag tagToRemove)
        {
            return this.Tags.Remove(tagToRemove);
        }

        public bool AddReaction(Reaction reaction)
        {
            if (this.Reactions.Contains(reaction))
                return false;

            this.Reactions.Add(reaction);
            return true;
        }

        public bool RemoveReaction(Reaction reaction)
        {
            return this.Reactions.Remove(reaction);
        }

        public bool AddComment(Comment comment)
        {
            if (this.Comments.Contains(comment))
                return false;

            this.Comments.Add(comment);
            return true;
        }

        public bool RemoveComment(Comment comment)
        {
            return this.Comments.Remove(comment);
        }

        public override bool Equals(object obj)
        {
            if (obj == this)
                return true;

            if (obj.GetType() != typeof(Post))
                return false;

            Post otherPost = (Post)obj;

            return otherPost.Id.Equals(this.Id);
        }

        public override int GetHashCode()
        {
            return HashCode.Combine(this.Id);
        }
    }

}

File: SocialNetworkGame/PostMasterData/src/domain/post.ts
import { AggregateRoot } from "../core/domain/AggregateRoot";
import { Reaction } from "./reaction"
import { Comment } from "./comment"
import { UniqueEntityID } from "../core/domain/UniqueEntityID";
import { Result } from "../core/logic/Result";
import { Guard } from "../core/logic/Guard";


interface PostProps {
    postText: string;
    reactions: Reaction[];
    comments: Comment[];
    creationDate: string;
    playerCreator: string;
    tags: string[];

}


export class Post extends AggregateRoot<PostProps> {
    get id (): UniqueEntityID {
        return this._id;
    }

    get postText(): string {
      return this.props.postText;
    }

    get reactions(): Reaction[] {
      return this.props.reactions;
    }

    get comments(): Comment[] {
      return this.props.comments;
    }

    get creationDate(): string {
      return this.props.creationDate;
    }

    get playerCreator(): string {
      return this.props.playerCreator;
    }

    get tags(): string[] {
      return this.props.tags;
    }


    private constructor (props: PostProps, id?: UniqueEntityID) {
      super(props, id);
    }

    public static create(props: PostProps, id?: UniqueEntityID): Result<Post> {
      const guardedProps = [
        { argument: props.postText, argumentName: 'postText' },
        { argument: props.reactions, argumentName: 'reactions' },
        { argument: props.comments, argumentName: 'comments' },
        { argument: props.creationDate, argumentName: 'creationDate' },
        { argument: props.playerCreator, argumentName: 'playerCreator' },
        { argument: props.tags, argumentName: 'tags' }
      ];

      const guardResult = Guard.againstNullOrUndefinedBulk(guardedProps);

      if (!guardResult.succeeded) {
        return Result.fail<Post>(guardResult.message)
      }
      else {
        const post = new Post({
          ...props
        }, id);

        return Result.ok<Post>(post);
      }
    }

    public assignTag(tag: string): boolean {
      if (this.props.tags.includes(tag)) {
        return false;
      }
      this.props.tags.push(tag);
      return true;
    }

    public removeTag(tag: string): void {
      const index = this.props.tags.indexOf(tag, 0);
      if (index > -1) {
        this.props.tags.splice(index, 1);
      }
    }

    public addComment(comment: Comment): boolean {
      if (this.props.comments && this.props.comments.includes(comment)) {
        return false;
      }

      if (!this.props.comments)
        this.props.comments = []

      this.props.comments.push(comment);
      return true;
    }

    public removeComment(comment: Comment): void {
      const index = this.props.comments.indexOf(comment, 0);
      if (index > -1) {
        this.props.comments.splice(index, 1);
      }
    }
    public addReaction(reaction:Reaction):boolean{
      if (this.props.reactions && this.props.reactions.includes(reaction)) {
        return false;
      }
      if (!this.props.comments)
        this.props.reactions=[]

      this.props.reactions.push(reaction)
      return true;
    }

    public removeReaction(reaction:Reaction):void{
      const index=this.props.reactions.indexOf(reaction, 0);
      if (index>-1){
        this.props.reactions.splice(index, 1);
      }
    }

    // TODO addReaction, removeReaction

}


